GPSTest/src/main/java/com/android/gpstest/util/MathUtils.java

```diff
@@ -124,6 +128,20 @@ public static float mapToRange(float a, float minA, float maxA,
         return bShifted + minB;
     }
 
+    public static HashMap<String, Integer> getDMSFromLocation(Double location) {
```

Comment:
Hehe, I started out doing just that. But I couldn't find out how to avoid passing the context to get mRes, and importing R to get the gps_latitude_dms_value string from strings.xml...so I opted for a math-only function and left the string formatting code in GpsStatusFragment.

So -- would it be okay to just remove gps_latitude_dms_value string from string.xml and just hard-code the string-to-be-formatted in UIUtils, and then just use the generic printf formatter function? I don't think there will ever be localization support required for DMS, and that wouldn't add a lot of dependencies into UIUtils. If that's a good approach let me know and I'll do it when I get time.

In addition -- I could encapsulate the preference-check in the UIUtils function and just return the location as a string, whether decimal or DMS. That will simplify GpsStatusFragment and the decision of the format of the location seems like it should be entirely in the hands of a UIUtils function, anyways. Let me know if this is reasonable.

Thanks for the feedback!

Reply:
Yeah, the Context issue is omnipresent.  For better or worse, the solution I settled on with GPSTest is to have a static method `Application.get()` that allows fetching the Application context when using a Activity or Fragment context may be dangerous due to lifecycle issues.  This block does fall within the code path of `onLocationChanged()`, which is an async callback from the platform, so using Application.get() seems reasonable here (an aside - I prefer to use Application.get()  in utility methods vs. passing in Context as it's too easy to accidentally use the method somewhere with an Activity or Fragment context that causes lifecycle issues).

>So -- would it be okay to just remove gps_latitude_dms_value string from string.xml and just hard-code the string-to-be-formatted in UIUtils, and then just use the generic printf formatter function? I don't think there will ever be localization support required for DMS, and that wouldn't add a lot of dependencies into UIUtils. If that's a good approach let me know and I'll do it when I get time.

This reasoning seems valid, although I'll confess I don't recall off the top of my head how printf handles Unicode and if these characters are in fact localized.  Let me know what you think/find - if some uncertainty exists, it seems like using Application.get() for the context would be the safest approach.

>In addition -- I could encapsulate the preference-check in the UIUtils function and just return the location as a string, whether decimal or DMS. That will simplify GpsStatusFragment and the decision of the format of the location seems like it should be entirely in the hands of a UIUtils function, anyways. Let me know if this is reasonable.

Good idea - sounds good to me.